source/robot_model/include/robot_model/Model.hpp

```diff
@@ -261,6 +276,42 @@ class Model {
    * @return the list of parameters
    */
   std::list<std::shared_ptr<state_representation::ParameterInterface>> get_parameters() const;
+
+  /**
+   * @brief It checks if the joint positions are inside at their limits
+   * @param joint_positions the joint positions to check
+   * @return true if the positions are inside their limits, false otherwise.
+   */
+  bool in_range(const state_representation::JointPositions& joint_positions);
+
+  /**
+   * @brief It checks if the joint velocities are inside at their limits
+   * @param joint_velocities the joint velocities to check
+   * @return true if the velocities are inside their limits, false otherwise.
+   */
+  bool in_range(const state_representation::JointVelocities& joint_velocities);
+
+  /**
+   * @brief It checks if the joint torques are inside at their limits
+   * @param joint_torques the joint torques to check
+   * @return true if the torques are inside their limits, false otherwise.
+   */
+  bool in_range(const state_representation::JointTorques& joint_torques);
+
+  /**
+   * @brief It checks if the joint state is inside at its limits
+   * @param joint_states the joint state to check
+   * @return true if the state is inside its limits, false otherwise.
+   */
+  bool in_range(const state_representation::JointState& joint_states);
```

Comment:
@eeberhard @buschbapti I already wanted to propose to @patr3 to rename `joint_states` to `joint_state` for consistency but I realized that the naming of the classes is not very consistent already.. (plural for single state variables like `JointPositions` but singular for a joint state `JointState` even though this includes several joints too). Should we do something about this?

Reply:
I was initially seeing a state as one state of all the joints not one state per joint hence the singular for `JointState`, but containing multiple joint positions, velocities, etc... It is indeed disputable. Given the current logic, however, I would argue indeed to remove the `s` at `joint_states` unless you want to provide a vector of `JointState`.

Reply:
This is also consistent with ROS naming convention (one `JointState` http://docs.ros.org/en/melodic/api/sensor_msgs/html/msg/JointState.html)

Reply:
Agreed!